File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for(int i = 0; i < nums.size(); i++) {
            int currentNum = nums[i];
            int comp = target - nums[i];
            if(seen.contains(comp)) {
                return {seen[comp], i};
            }
            seen[currentNum] = i;
        }

        return {0,0};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i, x in enumerate(nums):
            comp = target - x
            if comp in d :
                return d[comp], i
            d[x] = i
        return {}
        
            